carbon-design-system/carbon-addons-iot-react
# fix(hotspoteditormodal): handle element content in hotspots

Closes #2384

**Summary**
The Hotspoteditormodal now handles cases where the content of the hotspot is a react element. It does that by disabling the sections of the UI that would conflict with the content property of the hotspot, but leaves the rest of the hotspot configuration editable.

**Note!**
Same issue here as for the latest PRs. There are updated snapshots from a previous commit.

**Change List (commits, features, bugs, etc)**
Colordropdown:
- Added disabled prop to 
- updated story to show new prop
- added unit tests
- renamed and fixed test-id prop

HotspoteditorModal:
- updated story with hotspots that has an element as content, both text and fixed
- disabled data source tab when selected hotspot content is an element 
- added unit tests 

HotspotEditorTooltipTab:
- disabled title and description input when selected hotspot content is an element 
- added unit tests 

HotspotTextStyleTab:
- disabled font color and fontsize input when selected hotspot content is an element 
- added unit tests 

**Acceptance Test (how to verify the PR)**
1. Go to https://deploy-preview-2427--carbon-addons-iot-react.netlify.app?path=/story/2-watson-iot-experimental-☢%EF%B8%8F-hotspoteditormodal--width-existing-hotspots
2. Select the green hotspot and make sure DataSourceTab, Title and Description are disabled
3. Verify that the other configurations can be changed as normal
4. Select the white text-based hotspot "content is an element" and make sure  DataSourceTab and all font related settings are disabled. 
5. Verify that the other configurations can be changed as normal
6. Repeat the steps above for "fixed" and "text/label" hotspots that you create yourself

**Regression Test (how to make sure this PR doesn't break old functionality)**
Verify that the other HotspotEditorModal stories work
